docs/src/architecture.md

```
@@ -0,0 +1,344 @@
+  Kustomize["Kustomize"]:::external
+  Sentry["Sentry"]:::external
+end
+
+Reconciler -->|uses| K8s
+Main --> Kustomize & Sentry
+
+```
+
+## Resolver Architecture
+
+
```

markdownlint-cli2 (0.17.2)

Multiple consecutive blank lines
Expected: 1; Actual: 2

(MD012, no-multiple-blanks)